src/VisualStudio/Core/Def/Implementation/ProjectSystem/CPS/ITempPECompiler.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.LanguageServices.ProjectSystem
+{
+    /// <summary>
+    /// Provides TempPE compiler access
+    /// </summary>
+    /// <remarks>
+    /// This is used by the project system to enable designer support in Visual Studio
+    /// </remarks>
+    internal interface ITempPECompiler
+    {
+        /// <summary>
+        /// Compiles specific files into the TempPE DLL to provide designer support
+        /// </summary>
+        /// <param name="context">The project context.</param>
+        /// <param name="outputFileName">The binary output path.</param>
+        /// <param name="filesToInclude">Array of file paths from the project that should be included in the output.</param>
+        /// <param name="cancellationToken">The cancellation token</param>
+        /// <returns><see langword="true" /> if the compilation was successful</returns>
```

Comment:
"succeeded without build errors" is that you mean?

Reply:
This leaks I/O exceptions out of it; this should either be documented or we should handle those and return "False".

Reply:
> "succeeded without build errors" is that you mean?

I'm not checking for errors so I think what I have here is more correct. I'm just copying what the existing TempPE compiler did, checking Success only
https://github.com/dotnet/roslyn/blob/fab7134296816fc80019c60b0f5bef7400cf23ea/src/VisualStudio/CSharp/Impl/ProjectSystemShim/TempPECompilerService.cs#L65

Reply:
I've documented the exception and will handle it on the project system side.